### PR TITLE
new feature: human in the loop auto labeling

### PR DESCRIPTION
**Pull request by yongbing-chen**

### CODE REVIEW

**Comment by alex-krasn** [comment]
There are three different colors for tag icons in documents preview... the green one does not have a tooltip with explanation - what the color mean, when purpure and brown ones do.
image

**Comment by alex-krasn** [comment]
I will suggest for brown tag: "Manually labeled, not revised"; for purpure:  "
Automatically labeled", for green one: "Visited/Revised".

**Comment by alex-krasn** [comment]
After user visit documents with brown icon, it become green (manually labeled and revised) - but if the user loses focus (switch to another window), visited documents icons turns brown again.

**Comment by alex-krasn** [comment]
I'm still gettin state mutation error (rarely, though)
image

File: src/react/components/common/tagInput/tagInputItem.tsx [hunk 1]
```diff
@@ -107,6 +113,7 @@ export default class TagInputItem extends React.Component<ITagInputItemProps, IT
                     }
                     {this.renderTagDetail()}
                 </div>
+
```
Comment (alex-krasn) [inline src/react/components/common/tagInput/tagInputItem.tsx:116]: do we really need empty line here?

File: src/react/components/pages/editorPage/canvasCommandBar.tsx [hunk 2]
```diff
@@ -64,16 +76,6 @@ export const CanvasCommandBar: React.FunctionComponent<ICanvasCommandBarProps> =
...
-                    // {
-                    //   key: "DrawnRegions",
-                    //   text: strings.editorPage.canvas.canvasCommandBar.items.layers.subMenuItems.drawnRegions,
-                    //   canCheck: true,
-                    //   iconProps: { iconName: "AddField" },
-                    //   isChecked: props.layers["drawnRegions"],
-                    //   className: props.drawRegionMode ? "disabled" : "",
-                    //   onClick: () => props.handleLayerChange("drawnRegions"),
-                    //   disabled: props.drawRegionMode
-                    // },
```
Comment (stew-ro) [inline src/react/components/pages/editorPage/canvasCommandBar.tsx:76]: Don't delete these. We're leaving them uncommented until back-end is ready

File: src/react/components/pages/editorPage/canvasCommandBar.tsx [hunk 3]
```diff
@@ -85,16 +87,6 @@ export const CanvasCommandBar: React.FunctionComponent<ICanvasCommandBarProps> =
...
-            // {
-            //   key: "drawRegion",
-            //   text: strings.editorPage.canvas.canvasCommandBar.items.drawRegion,
-            //   iconProps: { iconName: "AddField" },
-            //   toggle: true,
-            //   checked: props.drawRegionMode,
-            //   className: !props.layers["drawnRegions"] ? "disabled" : "",
-            //   onClick: () => props.handleToggleDrawRegionMode(),
-            //   disabled: !props.layers["drawnRegions"],
-            // }
```
Comment (stew-ro) [inline src/react/components/pages/editorPage/canvasCommandBar.tsx:97]: Don't delete

Comment (yongbing-chen) [reply src/react/components/pages/editorPage/canvasCommandBar.tsx:97]: add the code back

File: src/react/components/pages/editorPage/canvasCommandBar.tsx (hunk 2 above)
Comment (yongbing-chen) [reply src/react/components/pages/editorPage/canvasCommandBar.tsx]: add the code back

**Review by stew-ro — DISMISSED** [review]

**Review by stew-ro — APPROVED** [review]

**Pull request merged**